charts/spark-operator-chart/tests/controller/service_test.yaml

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2024 The Kubeflow authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+suite: Test controller deployment
+
+templates:
+  - controller/service.yaml
+
+release:
+  name: spark-operator
+  namespace: spark-operator
+
+tests:
+  - it: Should create the empty service correctly
+    asserts:
+      - containsDocument:
+          apiVersion: v1
+          kind: Service
+          name: spark-operator-controller-svc
+      - isNullOrEmpty:
+          path: spec.ports
```

Comment:
The [e2e test CI](https://github.com/kubeflow/spark-operator/actions/runs/10847662027/job/30103203369?pr=2164) has failed for that the controller service cannot be created with empty port lists. I think we should not create service for controller if pprof is disabled.